### PR TITLE
unique numbers, n queen with bitmask in c, codehef snugfit and hardcash and matches

### PR DESCRIPTION
**Pull request by Ankit152**
**Fixes issue:**

#4707
#4723 
#4765 
#4766 
#4868 
#4869 

**Changes:**

1. Unique numbers in an array using STL function in O(1) space and O(NlogN) time.
2. Implementation of N-Queen problem in C using bitmasking.
3. Solution to codechef problem SNUGFIT in C++.
4. Solution to codechef problem HARDCASH in C++.
5. Created a folder having EIDI2 C++ code and Readme file.
6. Implemented matches code in C and C++ with a readme file.

### CODE REVIEW

File: code/backtracking/src/n_queen/nqueen_bitmask.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include <stdio.h>
+
+long long int n, ans = 0, DONE;
```
Comment (arnavb) [inline code/backtracking/src/n_queen/nqueen_bitmask.c]: n should be local to main. ans should be local to `solve` and returned. When making the changes I say, do not create a new PR. Just push to the same branch to update.

File: code/backtracking/src/n_queen/nqueen_bitmask.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  if (rowMask == DONE) 
+  {
+    ans++;
+    return;
+  }
+  long long int safe = DONE & (~(rowMask | ld | rd));
+  while (safe) 
+  {
+    long long int p = safe & (-safe);
+    safe = safe - p;
```
Comment (arnavb) [inline code/backtracking/src/n_queen/nqueen_bitmask.c]:
```suggestion
    safe -= p;
```

File: code/backtracking/src/n_queen/nqueen_bitmask.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    solve(rowMask | p, (ld | p) << 1, (rd | p) >> 1);
+  }
+}
+int main() 
+{
+  scanf("%lld", &n);
+  DONE = (1 << n) - 1;
```
Comment (arnavb) [inline code/backtracking/src/n_queen/nqueen_bitmask.c]: Should be completely lowercase variable name.

File: code/online_challenges/src/codechef/CASH/hard_cash.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#include <iostream>
```
Comment (arnavb) [inline code/online_challenges/src/codechef/CASH/hard_cash.cpp]: Reformat both files using 4 space indentation.

File: code/online_challenges/src/codechef/CASH/hard_cash.cpp [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+using namespace std;
```
Comment (arnavb) [inline code/online_challenges/src/codechef/CASH/hard_cash.cpp]: Don't use this.

File: code/online_challenges/src/codechef/CASH/hard_cash.cpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+int main() 
+{
+  int t;
+  cin >> t;
+  while (t--) {
+    int n, k, x, sum = 0;
+    cin >> n >> k;
+    for (int i = 0; i < n; i++)
```
Comment (arnavb) [inline code/online_challenges/src/codechef/CASH/hard_cash.cpp]:
```suggestion
    for (int i = 0; i < n; ++i) 
```

File: code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp [hunk 7]
```diff
@@ -0,0 +1,31 @@
+#include <iostream>
+
```
Comment (arnavb) [inline code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp]:
```suggestion
```

File: code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+#include <algorithm>
+
+using namespace std;
```
Comment (arnavb) [inline code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp]: Don't use this.

File: code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+int main() 
+{
+  int t;
+  cin >> t;
+  while (t--) 
+  {
+    int n, i, sum = 0;
+    cin >> n;
+    int a[n], b[n];
```
Comment (arnavb) [inline code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp]: This is invalid C++. Just `std::vector` instead.

File: code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    for (i = 0; i < n; i++)
```
Comment (arnavb) [inline code/online_challenges/src/codechef/SNUG_FIT/snug_fit.cpp]: Preincrement in both loops.

File: code/unclassified/src/unique_number/unique_num_stl.cpp [hunk 11]
```diff
@@ -0,0 +1,25 @@
+#include <iostream>
+
+#include <vector>
+
```
Comment (arnavb) [inline code/unclassified/src/unique_number/unique_num_stl.cpp]:
```suggestion
```

File: code/unclassified/src/unique_number/unique_num_stl.cpp [hunk 12]
```diff
@@ -0,0 +1,25 @@
+#include <iostream>
+
```
Comment (arnavb) [inline code/unclassified/src/unique_number/unique_num_stl.cpp]:
```suggestion
```

File: code/unclassified/src/unique_number/unique_num_stl.cpp [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+#include <vector>
+
+#include <algorithm>
+
+using namespace std;
```
Comment (arnavb) [inline code/unclassified/src/unique_number/unique_num_stl.cpp]: Same as before.

Comment (Ankit152) [reply code/unclassified/src/unique_number/unique_num_stl.cpp]: can you help we to understand what is clang format?

Comment (arnavb) [reply code/unclassified/src/unique_number/unique_num_stl.cpp]: @Ankit152 Use [this website](https://zed0.co.uk/clang-format-configurator/) paste your code in, set indent width to 4 spaces, and reformat your code.

**Comment by Ankit152** [comment]
@arnavb @AdiChat please review my PR.

**Comment by arnavb** [comment]
Remove all instances of `using namespace std`. Also make sure all files end in a newline.

**Comment by Ankit152** [comment]
sir can you please explain me what "all files end in a new line" means? i am not getting it.

**Comment by Ankit152** [comment]
@arnavb @AdiChat sir please review my PR.

**Pull request merged**